src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -909,9 +912,12 @@ void IdealGraphPrinter::update_compiled_method(ciMethod* current_method) {
   assert(C != nullptr, "must already be set");
   if (current_method != _current_method) {
     // If a different method, end the old and begin with the new one.
-    end_method();
-    _current_method = nullptr;
-    begin_method();
+    if (_append) {
+      _current_method = current_method;
+    } else {
+      end_method();
+      begin_method();
+    }
```

Comment:
Could you explain this change and maybe add a comment?

Reply:
Reason for adding `if (_append)`: in append mode, the constructor of `IdealGraphPrinter` will not call `begin_method`, so we should not call `end_method` here to prevent the assertion failure, just update `_current_method`.

Reason for removing `_current_method = nullptr;`: `_current_method` is maintained and cleared in `end_method`, we do not need to handle it manually here.

Added more comments in the latest commit.

Reply:
Thanks!